modules/audigentRtdProvider.js

```diff
@@ -11,24 +11,54 @@ import {submodule} from '../src/hook.js';
 import {ajax} from '../src/ajax.js';
 import { getStorageManager } from '../src/storageManager.js';
 
-const storage = getStorageManager();
+export const storage = getStorageManager();
 
 /** @type {string} */
 const MODULE_NAME = 'realTimeData';
 const SUBMODULE_NAME = 'audigent';
-const HALOID_LOCAL_NAME = 'auHaloId';
-const SEG_LOCAL_NAME = '__adgntseg';
+export const HALOID_LOCAL_NAME = 'auHaloId';
+export const SEG_LOCAL_NAME = '__adgntseg';
+
+const set = (obj, path, val) => {
+  const keys = path.split('.');
+  const lastKey = keys.pop();
+  const lastObj = keys.reduce((obj, key) => obj[key] = obj[key] || {}, obj);
+  lastObj[lastKey] = lastObj[lastKey] || val;
+};
+
+/** bid adapter format segment augmentation functions */
+const segmentMappers = {
+  appnexus: function(bid, segments) {
+    set(bid, 'params.user.segments', []);
+    bid.params.user.segments = bid.params.user.segments.concat(segments);
+  },
+  generic: function(bid, segments) {
+    bid.segments = segments;
```

Comment:
I think `bid.fpd.user.data.segments` should be set for each bidder. The long term goal would be for per-bidder differences to go away. It's understandable that working with existing bidder segment locations is desirable, but if we establish a convention, we can encourage bidders to migrate to the convention over time.

Reply:
Fully agree that we should focus on the fpd bidder integrations.  What we can do is remove individual bidders from our module once they have integrated fpd, and work together with them if pubs have interest in using them with our module to integrate.